wforce/dump_entries.cc

```diff
@@ -0,0 +1,190 @@
+#include <fstream>
+#include "iputils.hh"
+#include "json11.hpp"
+#include "base64.hh"
+#include "sstuff.hh"
+
+using namespace std;
+using namespace json11;
+
+void print_help()
+{
+  cout<<"Syntax: dump_entries [-u,--uri wforce uri] [-l,--local-addr address] [-p,--local-port port] [-w,--wforce-pwd password] [-f,--filename file] [-h,--help]\n";
+  cout<<"\n";
+  cout<<"-u,--uri address          <Mandatory> Connect to wforce on the specified URI\n";
+  cout<<"-l,--local-addr address   <Mandatory> Listen for results on the specified local IP address\n";
+  cout<<"-p,--local-port port      <Mandatory> Listen on the specified port\n";
+  cout<<"-w,--wforce-pwd password  <Mandatory> The password to authenticate to wforce\n";
+  cout<<"-f,--filename file        <Optional>  Send the results to a file\n";
+  cout<<"-h,--help                 Display this helpful message\n";
+  cout<<"\n";
+}
+
+bool g_console = true;
+
+int main(int argc, char** argv)
+{
+  string uri, local_addr, password, filename;
+  int local_port=0;
+  struct option longopts[]={ 
+    {"uri", required_argument, 0, 'u'},
+    {"local-addr", required_argument, 0, 'l'},
+    {"local-port", required_argument, 0, 'p'},
+    {"wforce-pwd", required_argument, 0, 'w'},
+    {"filename", required_argument, 0, 'f'},
+    {"help", 0, 0, 'h'},
+    {0,0,0,0}
+  };
+  int longindex=0;
+  for(;;) {
+    int c=getopt_long(argc, argv, "hu:l:p:w:f:", longopts, &longindex);
+    if(c==-1)
+      break;
+    switch(c) {
+    case 'u':
+      uri = optarg;
+      break;
+    case 'l':
+      local_addr = optarg;
+      break;
+    case 'p':
+      try {
+        local_port = stoi(optarg);
+      }
+      catch (const std::exception& e) {
+        cerr << "Could not parse local port - must be an integer" << endl;
+        exit(EXIT_FAILURE);
+      }
+      break;
+    case 'w':
+      password = optarg;
+      break;
+    case 'f':
+      filename = optarg;
+      break;
+    case 'h':
+      print_help();
+      exit(EXIT_SUCCESS);
+      break;
+    }
+  }
+  argc-=optind;
+  argv+=optind;
+
+  if ((uri.empty()) ||
+      (local_addr.empty()) ||
+      (local_port == 0) ||
+      (password.empty())) {
+    cout << "Missing mandatory arguments...\n";
+    print_help();
+    exit(EXIT_FAILURE);
+  }
+
+  // Open file if specified
+  ofstream myfile;
+  if (!filename.empty()) {
+    myfile.open(filename, ios::out | ios::trunc);
+
+    if (!myfile.is_open()) {
+      cerr << "Could not open " << filename << " for writing." << endl;
+      exit(EXIT_FAILURE);
+    }
+  }
+  
+  // Listen on local IP and port
+  ComboAddress local_ca;
+  try {
+    local_ca = ComboAddress(local_addr, local_port);
+  }
+  catch (const WforceException& e) {
+    cerr << "Error parsing local address and port " << local_addr << ", " << local_port << ". Make sure to use IP addresses not hostnames" << endl;
+    exit(EXIT_FAILURE);
+  }
+  
+  int listen_sock = socket(local_ca.sin4.sin_family, SOCK_STREAM, 0);
```

Comment:
Is it closed ?

Reply:
Hmm, not by the looks of it.

Reply:
Wrapped in a Socket class to auto-close it